main.py

```diff
@@ -16,23 +17,28 @@ def clean():
         config.load_kube_config()
 
     v1api = client.CoreV1Api()
-    #v1beta1api = client.AppsV1beta1Api()
-    #max_namespace_inactive_days = os.environ['MAX_NAMESPACE_INACTIVE_HOURS']
+    v1beta1api = client.AppsV1beta1Api()
+    max_namespace_inactive_days = os.environ['MAX_NAMESPACE_INACTIVE_HOURS']
+
+    ns_whitelist = os.environ['NS_WHITELIST'].split(',')
+
+    stale = conditions.AND(InactiveDeploymentCondition(v1beta1api, max_namespace_inactive_days),
+                           conditions.NotWhitelisted(ns_whitelist))
 
     # reading environment variable
     vsts_token = os.environ['VSTS_PAT']
+
     cleanup_conditions = [
         AnnotationAllowCleanupIsTrueCondition(),
-        #InactiveDeploymentCondition(v1beta1api, max_namespace_inactive_days),
-        VSTSRefDeletedCondition(vsts_token),
+        conditions.OR(VSTSRefDeletedCondition(vsts_token), stale)
     ]
```

Comment:
Should we use conditions.AND in here?

Reply:
conditions.AND(AnnotationAllowCleanupIsTrueCondition(), conditions.OR(VSTSRefDeletedCondition(vsts_token), stale))

Reply:
hmm guess we could remove all all then. let me give it a shot.

Reply:
and is wierd derivative of all/apply and OR is a derivate of any/apply

Reply:
As you demoted in your next comments :)